pkg/syncer/spec/mutators/deployment_test.go

```diff
@@ -510,6 +609,45 @@ func TestMutate(t *testing.T) {
 	}
 }
 
+func TestPrefixSecretName(t *testing.T) {
+	for _, c := range []struct {
+		desc         string
+		name         string
+		expectedName string
+	}{
+		{
+			desc:         "short name, we just expect it to be prefixed",
+			name:         "test-secret",
+			expectedName: "kcp-test-secret",
+		},
+		{
+			desc:         "long name of 253, expect a prefixed valid name of 253 characters",
+			name:         "ke2wlbx5hrmx41fu0ak55a3d0p81dasqcbzka0fz0p50t572aimct5k7d3f029196g5ryx54je6v09evvccclblg2aqdhvkfemkfvxxhrn66wj6hzvgfsmdb79bqaavmdp5kqzx8bx90pwp9n8hazvqzrbcbyrx77kyldyzacoqh7e7o2tvz3eja9spjn2f5opyxg0ui804548v9dh03zbgt57p1ddbeur8z6z017eu4vmff8kppj79ugkxbs",
+			expectedName: "kcp-ke2wlbx5hrmx41fu0ak55a3d0p81dasqcbzka0fz0p50t572aimct5k7d3f029196g5ryx54je6v09evvccclblg2aqdhvkfemkfvxxhrn66wj6hzvgfsmdb79bqaavmdp5kqzx8bx90pwp9n8hazvqzrbcbyrx77kyldyzacoqh7e7o2tvz3eja9spjn2f5opyxg0ui804548v9dh03zbgt57p1ddbeur8z6z017eu4vmff8k-c08535",
+		},
+	} {
+		{
+			t.Run(c.desc, func(t *testing.T) {
+				actual := PrefixSecretName(c.name)
+				if actual != c.expectedName {
+					t.Errorf("expected secret names are not equal, got:\n %#v \n wanted:\n %#v \n", actual, c.expectedName)
+				}
+				if len(actual) > validation.DNS1123SubdomainMaxLength {
+					t.Errorf("secret name is longer than %d characters, got:\n %#v \n", validation.DNS1123SubdomainMaxLength, actual)
+				}
+			})
+		}
+	}
+}
+
+func toObjects(secrets []corev1.Secret) []runtime.Object {
+	objs := make([]runtime.Object, 0, len(secrets))
+	for _, secret := range secrets {
+		objs = append(objs, secret.DeepCopy())
```

Comment:
no random deepcopy in some helper please

Reply:
done